history/storage.py

```diff
@@ -187,45 +227,47 @@ def store_response(self, spider, request, response):
         s3_key = self.s3_bucket.new_key(key)
 
         try:
-            #s3_key.update_metadata(metadata) #=> can't use this as need to cast to unicode
+            metadata = _truncate_metadata_fields(metadata)
             for k, v in metadata.items():
-                if isinstance(v, str):
-                    v = v[:400] + '...' if len(v) > 400 else v
-                s3_key.set_metadata(k, unicode(v))
+                s3_key.set_metadata(k, v)
             s3_key.set_contents_from_string(data_string)
+            # save source file
+            job_folder = self.SAVE_SOURCE
+            # if the S3 key is too long, the AWS interface does not allow to download the file !
+            source_url = request.url[:200] + '...' if len(request.url) > 200 else request.url
```

Comment:
This time the comment makes it much better to understand but now we have an undocumented, hard-coded, 200 max length ?